corehq/motech/openmrs/atom_feed.py

```diff
@@ -284,3 +284,58 @@ def update_patient(repeater, patient_uuid, updated_at):
             xmlns=XMLNS_OPENMRS,
             device_id=OPENMRS_ATOM_FEED_DEVICE_ID + repeater.get_id,
         )
+
+
+def import_encounter(repeater, encounter_uuid, updated_at):
+    # It's possible that an OpenMRS concept appears more than once in
+    # form_configs. Use a defaultdict(list) so that earlier definitions
+    # don't get overwritten by later ones:
+
+    def fields_from_observations(observations, mappings):
+        """
+        Traverse a tree of observations, and return the ones mapped to
+        case properties.
+        """
+        fields = {}
+        for obs in observations:
+            if obs['concept']['uuid'] in mappings:
+                for mapping in mappings[obs['concept']['uuid']]:
+                    fields[mapping.case_property] = mapping.value.deserialize(obs['value'])
+            if obs['groupMembers']:
+                fields.update(fields_from_observations(obs['groupMembers'], mappings))
+        return fields
+
+    response = repeater.requests.get(
+        '/ws/rest/v1/bahmnicore/bahmniencounter/' + encounter_uuid,
+        {'includeAll': 'true'},
+        raise_for_status=True
+    )
+    encounter = response.json()
+
+    observation_mappings = defaultdict(list)
+    for form_config in repeater.openmrs_config.form_configs:
+        for obs_mapping in form_config.openmrs_observations:
+            if obs_mapping.value.check_direction(DIRECTION_IMPORT) and obs_mapping.case_property:
+                observation_mappings[obs_mapping.concept].append(obs_mapping)
```

Comment:
nothing in this function changes the mapping, could this chunk be moved to `repeater.openmrs_config` and cached so you don't have to do it for each encounter being imported?